src/commonMain/kotlin/org/cru/godtools/tool/state/State.kt

```diff
@@ -1,23 +1,28 @@
 package org.cru.godtools.tool.state
 
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.conflate
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.onStart
 import org.cru.godtools.tool.internal.Parcelable
 import org.cru.godtools.tool.internal.Parcelize
-import org.cru.godtools.tool.model.EVENT_NAMESPACE_STATE
-import org.cru.godtools.tool.model.EventId
 
 @Parcelize
-class State internal constructor(private val state: MutableMap<String, List<String>>) : Parcelable {
-    constructor() : this(mutableMapOf<String, List<String>>())
+class State internal constructor(private val state: MutableMap<String, List<String>?>) : Parcelable {
+    constructor() : this(mutableMapOf<String, List<String>?>())
 
-    operator fun get(key: String) = state[key]?.firstOrNull()
-    fun getAll(key: String) = state[key].orEmpty()
+    private val changeFlow = MutableSharedFlow<String>(extraBufferCapacity = Int.MAX_VALUE)
+    internal fun changeFlow(key: String) = changeFlow.filter { it == key }.onStart { emit(key) }.conflate()
```

Comment:
What is this ```changeFlow``` doing exactly? I know it's helping to manage the state of the multi-select but just curious how it's working under the hood. I notice it's a part of coroutines so I have somewhat of an idea of what's going on haha.

Reply:
https://kotlinlang.org/docs/flow.html

a `Flow` is a stream of values that something can `collect` and process each value in sequence.

the `changeFlow` emits a string every time the value for `key` is updated. This is then used by `Multiselect.Option` to monitor when the state of the `Multiselect` changes and re-evaluates `isSelected()` and returns that to anything that wants to monitor when an Option's `isSelected` changes (e.g. the UI displaying the option)